app/models/user.rb

```diff
@@ -1,5 +1,8 @@
 class User < ApplicationRecord
+  has_many :assignments
+  has_many :tasks, through: :assignments
   has_many :journals, dependent: :destroy, inverse_of: :user
+
```

Comment:
Can you try to make another association between user and task?
For example:
has_many :created_tasks ..... referring to the Task class and using creator_id as a foreign key?